### DISCUSSION TITLE
[dotnet] Why not use .Net Core

### DISCUSSION THREAD

**Discussion by wuliangtdi** [godotengine/godot-proposals#4390] [Scripting]
We know that .net core supports windows, liunx and macos, and the performance of .net core is better than Mono. I want to support .net core.

**Comment by GeorgeS2019** [comment]
@wuliangtdi  We skip .NET Core and straight to the latest [.NET6](https://github.com/godotengine/godot-proposals/issues/2333#issuecomment-1076175654) and later .NET7

Just a personal feedback, could you add [dotnet] to your title, to be consistent with how the Godot team label Godot issue => **[dotnet] Why not use .Net Core**

For Godot4 dotnet6 Linux @lewiji is the person from Godot community

**Reply by wuliangtdi** [reply]
So the next big version of Godot will use .NET 6 or .NET 7?

**Reply by GeorgeS2019** [reply]
# Godot4 .NET6 [in action](https://gist.github.com/lewiji/2b86f47dcb142a10e104504a343c38ff?permalink_comment_id=4106318#gistcomment-4106318) (Ubuntu 20.04)

https://user-images.githubusercontent.com/233380/159492329-823c6dad-06e7-42af-b31b-91047d9870a0.mp4


![image](https://user-images.githubusercontent.com/49812372/162927323-1052589f-2061-4977-87a8-574ba7a61ad7.png)

**Reply by wuliangtdi** [reply]
wow,it's so great.The performance of godot will be better.Does godot4 also use .net 6 on Windows?

**Reply by GeorgeS2019** [reply]
We can get[ Godot4 .NET6 to compile, run and display using MSVC on Windows 11](https://gist.github.com/lewiji/2b86f47dcb142a10e104504a343c38ff?permalink_comment_id=4104908#gistcomment-4104908), however there is still some bugs with MSVC. It can not create csproj project and attach a cs# scripts

We need more people to try build and report problems to the Godot team working on the dotnet6 branch for Windows.

The next step is to use LLVM-MinGW to cross compile from Linux to Windows. This could by pass the problem using MSVC (hopefully)

**Reply by wuliangtdi** [reply]
csharp will be compiled into machine code, and the performance will be improved a lot. I am looking forward to the arrival of godot4.thanks for your answer

**Reply by GeorgeS2019** [reply]
@wuliangtdi 
[How to compile](https://github.com/godotengine/godot-proposals/discussions/4416#dotnet6-msvc) Godot4 dotnet6 Windows using MSVC

**Comment by Delpire** [comment]
I would be a little specific about performance improvements by going from mono to .NET Core/5/6. The performance improvements will really only affect the code you write and the performance of _Process. Things like draw calls and physics calculations (which are often the biggest performance issues) likely won't be improved by going to .NET 6 as far as I understand it.

Regardless, I think there will be a huge amount of workflow improvements (easier debugging, performance monitoring, etc.), when .NET 6 is finally supported.

**Comment by GeorgeS2019** [comment]
@Delpire do consider providing feedback/suggestions on easier debugging, performance monitoring, etc. on [**[dotnet] Godot4: supporting the transition from mono to dotnet6**](https://github.com/godotengine/godot-proposals/discussions/4416#discussioncomment-2606448)
